org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractEditorTest.java

```diff
@@ -8,53 +8,60 @@
  *******************************************************************************/
 package org.eclipse.xtext.ui.testing;
 
+import java.io.File;
 import java.lang.reflect.Field;
 
+import org.eclipse.core.filesystem.EFS;
+import org.eclipse.core.filesystem.IFileStore;
 import org.eclipse.core.resources.IFile;
 import org.eclipse.core.runtime.Status;
 import org.eclipse.jface.text.source.ISourceViewer;
 import org.eclipse.jface.text.source.projection.ProjectionViewer;
 import org.eclipse.ui.IEditorPart;
-import org.eclipse.ui.PartInitException;
 import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.ide.FileStoreEditorInput;
 import org.eclipse.ui.internal.ErrorEditorPart;
 import org.eclipse.ui.part.FileEditorInput;
 import org.eclipse.xtext.ui.editor.XtextEditor;
 import org.eclipse.xtext.ui.editor.utils.EditorUtils;
 
 /**
  * @author Peter Friese - Initial contribution and API
+ * @author Arne Deutsch - Added support for external files
  * 
  * @since 2.12
  */
 public abstract class AbstractEditorTest extends AbstractWorkbenchTest {
 
 	static final long STEP_DELAY = 0;
-	
+
 	protected abstract String getEditorId();
 
 	protected XtextEditor openEditor(IFile file) throws Exception {
-		IEditorPart openEditor = openEditor(file, getEditorId());
+		return getXtextEditor(
```

Comment:
Nice. Can we add an openEditor with a URI, too? That would allow us to test with files in archives.

Reply:
Probably possible, althought I'm not sure if it would work in the end with more or less every URI or just for the archives case. But I would say in scope of this ticket this is not needed.

Reply:
In general we can open a stream for all kinds of URIs so a URI based input should do the trick. In any case you are right - this is not in the scope of this task.